src/plugins/ucx/ucx_backend.cpp

```diff
@@ -827,6 +828,63 @@ nixl_status_t nixlUcxEngine::prepXfer (const nixl_xfer_op_t &operation,
     return NIXL_SUCCESS;
 }
 
+// Estimate transfer cost using ucp_ep_query_cost
+nixl_status_t nixlUcxEngine::estimateXferCost (const nixl_xfer_op_t &operation,
+                                               const nixl_meta_dlist_t &local,
+                                               const nixl_meta_dlist_t &remote,
+                                               const std::string &agent_name,
+                                               nixlCostEstimate &estimate)
+{
+    if (local.descCount() != remote.descCount()) {
+        NIXL_ERROR << "Local (" << local.descCount() << ") and remote (" << remote.descCount()
+                   << ") descriptor lists differ in size for cost estimation";
+        return NIXL_ERR_MISMATCH;
+    }
+
+    if (local.descCount() == 0) {
+        // Nothing to do
+        estimate.setDuration(0);
+        return NIXL_SUCCESS;
+    }
+
+    double total_duration_sec = 0.0;
+    for (int i = 0; i < local.descCount(); i++) {
+        size_t lsize = local[i].len;
+        size_t rsize = remote[i].len;
+
+        nixlUcxPrivateMetadata *lmd = static_cast<nixlUcxPrivateMetadata*>(local[i].metadataP);
+        nixlUcxPublicMetadata *rmd = static_cast<nixlUcxPublicMetadata*>(remote[i].metadataP);
+
+        if (!lmd || !rmd) {
+            NIXL_ERROR << "Null metadata found in descriptor lists at index " << i << " during cost estimation";
+            return NIXL_ERR_INVALID_PARAM;
+        }
+
+        if (lsize != rsize) {
+            NIXL_ERROR << "Local size (" << lsize << ") != Remote size (" << rsize
+                       << ") at index " << i << " during cost estimation";
+            return NIXL_ERR_INVALID_PARAM;
+        }
+
+        double duration_sec = 0.0;
+        nixl_status_t ret = uw->estimateCost(rmd->conn.ep,    // Remote Endpoint
+                                             lmd->mem,        // Local buffer address
+                                             rmd->rkey,       // Remote buffer address
+                                             lsize,           // Message size
+                                             operation,       // NIXL operation type
+                                             duration_sec);   // Output duration
```

Comment:
i'd remove these comments, they do not seem to add value
also rkey is not a buffer address and mem is not a local  address